File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema17Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import dynabuffers.DynabuffersEngine
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class Schema17Test : AbstractDynabuffersTest() {

    lateinit var engine: DynabuffersEngine;

    @BeforeEach
    fun setUp() {
        engine = Dynabuffers.parse(Schema17Test::class.java.getResourceAsStream("/schema17.dbs"))
    }

    @Test
    fun namespaceTest() {
        val message = mapOf("contentA" to "abc")
        val result = engine.deserialize(engine.serialize(message, "request"), "request")

        assertMap(message, result)
    }

}

File: dynabuffers-python/tests/usecase/Schema17Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema17Test(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))

    def setUp(self):
        self.engine = Dynabuffers.parse(FileStream(self.root_dir + "/schema17.dbs"))

    def test_namespace(self):
        message = {"contentA":"abc"}
        result = self.engine.deserialize(self.engine.serialize(message, "request"), "request")

        self.assertEqual(message, result)


if __name__ == "__main__":
    unittest.main()
